### PR TITLE
fix: force object type for message trait's headers

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Force object type for message trait's headers in `2.0.0`, `2.1.0` and `2.2.0` versions.

PR for `2.3.0` https://github.com/asyncapi/spec-json-schemas/pull/155

**Related issue(s)**
Fixes https://github.com/asyncapi/spec-json-schemas/issues/82

### CODE REVIEW

**Comment by jonaslagoni** [comment]
@derberg @fmvilas @magicmatatjahu any progress here?

**Comment by magicmatatjahu** [comment]
@jonaslagoni You dug up some dead PR 😅  I'll make an update for 2.2 and a separate PR for 2.3.

**Comment by magicmatatjahu** [comment]
@derberg Could you check?

File: schemas/2.0.0.json [hunk 1]
```diff
@@ -869,12 +869,16 @@
           "type": "string"
         },
         "headers": {
-          "oneOf": [
+          "allOf": [
             {
-              "$ref": "#/definitions/Reference"
+              "$ref": "#/definitions/schema"
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply schemas/2.0.0.json:874]: By this https://github.com/asyncapi/spec-json-schemas/blob/486b7bb6a0d417d7e3d4cedc498d3948dadf14fc/schemas/2.0.0.json#L362 you can use `$ref`

[the remark this answers is not in the document]
Comment (fmvilas) [reply schemas/2.0.0.json:874]: Ahh true! JSON Schema has it defined 👍 Cool thanks!

**Review by fmvilas — APPROVED** [review]
LGTM. Let's wait for @derberg.

**Review by derberg — APPROVED** [review]

**Comment by fmvilas** [comment]
@magicmatatjahu feel free to merge it yourself :)

**Pull request merged**